python/lsst/daf/butler/datastores/remoteFileDatastore.py

```diff
@@ -0,0 +1,239 @@
+
+log = logging.getLogger(__name__)
+
+
+class RemoteFileDatastore(FileLikeDatastore):
+    """A datastore designed for files at remote locations.
+
+    Parameters
+    ----------
+    config : `DatastoreConfig` or `str`
+        Configuration. A string should refer to the name of the config file.
+    bridgeManager : `DatastoreRegistryBridgeManager`
+        Object that manages the interface between `Registry` and datastores.
+    butlerRoot : `str`, optional
+        New datastore root to use to override the configuration value.
+
+    Raises
+    ------
+    ValueError
+        If root location does not exist and ``create`` is `False` in the
+        configuration.
+
+    Notes
+    -----
+    Datastore supports non-link transfer modes for file-based ingest:
+    `"move"`, `"copy"`, and `None` (no transfer).
+    """
+
+    def __init__(self, config: Union[DatastoreConfig, str],
+                 bridgeManager: DatastoreRegistryBridgeManager, butlerRoot: str = None):
+        super().__init__(config, bridgeManager, butlerRoot)
+        if not self.root.exists():
+            try:
+                self.root.mkdir()
+            except ValueError as e:
+                raise ValueError(f"Can not create datastore root '{self.root}', check permissions.") from e
+
+    def _artifact_exists(self, location: Location) -> bool:
+        """Check that an artifact exists in this datastore at the specified
+        location.
+
+        Parameters
+        ----------
+        location : `Location`
+            Expected location of the artifact associated with this datastore.
+
+        Returns
+        -------
+        exists : `bool`
+            True if the location can be found, false otherwise.
+        """
+        log.debug("Checking if resource exists: %s", location.uri)
+        return location.uri.exists()
+
+    def _delete_artifact(self, location: Location) -> None:
+        """Delete the artifact from the datastore.
+
+        Parameters
+        ----------
+        location : `Location`
+            Location of the artifact associated with this datastore.
+        """
+        log.debug("Deleting file: %s", location.uri)
+        location.uri.remove()
+        log.debug("Successfully deleted file: %s", location.uri)
+
+    def _read_artifact_into_memory(self, getInfo: DatastoreFileGetInformation,
+                                   ref: DatasetRef, isComponent: bool = False) -> Any:
+        location = getInfo.location
+
+        log.debug("Downloading data from %s", location.uri)
+        serializedDataset = location.uri.read()
+
+        storedFileInfo = getInfo.info
+        if len(serializedDataset) != storedFileInfo.file_size:
+            raise RuntimeError("Integrity failure in Datastore. "
+                               f"Size of file {location.path} ({len(serializedDataset)}) "
+                               f"does not match recorded size of {storedFileInfo.file_size}")
```

Comment:
I recently added a function `unwrap` to `lsst.daf.butler.cli.utils`. Using it with one triple-quoted string might be more useful than many strings? It could be moved to a more general utils file. 
```
            raise RuntimeError(unwrap(f"""Integrity failure in Datastore. 
                                          Size of file {location.path} ({len(serializedDataset)})
                                          does not match recorded size of {storedFileInfo.file_size}""")
```